working/augmentation-libraries/feature-specification.md

```diff
@@ -1155,9 +1150,11 @@ extensionDeclaration ::=
 extensionBody ::= '{' (metadata classMemberDeclaration)* '}'
 
 extensionTypeDeclaration ::=
-  'augment'? 'extension' 'type' 'const'? typeIdentifier
-  typeParameters? representationDeclaration interfaces?
-  '{' (metadata classMemberDeclaration)* '}'
+    'extension' 'type' 'const'? typeIdentifier
+    typeParameters? representationDeclaration interfaces?
+    '{' (metadata classMemberDeclaration)* '}'
+  | 'augment' 'extension' 'type' typeIdentifier typeParameters? interfaces?
+    '{' (metadata classMemberDeclaration)* '}'
```

Comment:
There's a little more redundancy here than I would like. In addition to the one in `classDeclaration`, there are now three places where we have:

```
'{' (metadata classMemberDeclaration)* '}'
```

How about:

```
typeDeclarationBody ::= '{' (metadata classMemberDeclaration)* '}'

extensionTypeHeader ::= 
    'extension' 'type' 'const'? typeIdentifier typeParameters?
    representationDeclaration
  | 'augment' 'extension' 'type' typeIdentifier typeParameters?

extensionTypeDeclaration ::= extensionTypeHeader interfaces? typeDeclarationBody
```

And then update `classDeclaration` to use `typeDeclarationBody` too?

---

Alternatively, we could do:

```
extensionTypeDeclaration ::=
    'augment' 'extension' 'type' 'const'? typeIdentifier
    typeParameters? representationDeclaration interfaces?
    '{' (metadata classMemberDeclaration)* '}'
```

And then outside of the grammar say:

* It's a compile-time error if an extension type marked `augment` is marked `const`.
* It's a compile-time error if an extension type marked `augment` has a `representationDeclaration`.
* It's an compile-time error if an extension type not marked `augment` does not have a `representationDeclaration`.

(In practice, I suspect this is how the parser maintainers will implement it, so that they get better error recovery.)

---

Or just keep what you have. It's certainly not intolerable that way either.